### PR TITLE
Add templates for required sysctls max_map_count and arp_announce

### PR DESCRIPTION
**Pull request by dagrayvid**
**- What I did**
Add two files to templates/common/_base/files for setting sysctls vm.max_map_count and net.ipv4.all.arp_announce.

In OpenShift 4.13 we are planning to make the Node Tuning Operator an optional feature (composable OCP). Two sysctls currently set by NTO are needed for functional reasons, and should be set even in clusters where NTO is disabled.

vm.max_map_count=262144 is needed for OpenShift Logging to work (ElasticSearch), see [RH BZ#1793714](https://bugzilla.redhat.com/show_bug.cgi?id=1793714).
net.ipv4.conf.all.arp_announce=2 is needed for the OpenShift SDN (see [RH BZ#1758552](https://bugzilla.redhat.com/show_bug.cgi?id=1758552)).

NTO will remain enabled by default and will continue to apply recommended default tunings to improve scalability and performance, while providing the option for users to apply custom TuneD profiles.

**- How to verify it**
Ensure that /etc/sysctl.d/arp.conf and /etc/sysctl.d/vm-max-map.conf are created, and that the sysctl values are set correctly even when these sysctls are not set by NTO.

**- Description for the changelog**
Add templates for required sysctls max_map_count and arp_announce

/cc @jmencak

### CODE REVIEW

**Review by jmencak — COMMENTED** [review]
Thank you for the PR, David.  The changes look good to me.  Personally, I'd add a one-liner comment inside each of the new template files why the setting is needed with a link to the respective BZs though, so that it is immediately clear why the setting is needed without looking at the git history.

**Comment by dagrayvid** [comment]
> Thank you for the PR, David. The changes look good to me. Personally, I'd add a one-liner comment inside each of the new template files why the setting is needed with a link to the respective BZs though, so that it is immediately clear why the setting is needed without looking at the git history.

Thanks Jiri, I did this. I added the comments into the .conf files themselves, so that someone looking at the host OS will see the explanations.

**Review by jmencak — APPROVED** [review]
Thank you for the changes, David.
/lgtm
Needs approvals from the MCO team though.
/hold

**Comment by dagrayvid** [comment]
> 1. Is it possible for a cluster to go from having these unset to having them set (when upgrading from 4.12->4.13), and does that affect anything

In 4.12, NTO will be enabled and will by-default set these itself via the containerized TuneD daemon (operand). The only issue here is the one Jiri mentioned.  If a customer is using a custom TuneD profile that set's these to some other values, it will look like it no longer takes effect on 4.13, unless they have configured TuneD to override the settings from /etc/sysctl.d, so this should be noted in the release notes. 

> 2. for existing clusters with NTO as well, these will be presumably defined twice (which isn't an issue)

Correct, should not be an issue.

> 3. in 4.13, would you be able to remove NTO from an upgraded cluster? would that also delete associated MCs? Or is this only something you can set during install time?

My understanding is that you can only disable features like this at install time. You can enabled disabled-features after install, but [you can't disable the enabled-features later.](https://docs.openshift.com/container-platform/4.11/post_installation_configuration/cluster-capabilities.html) Regardless, I don't think this would have an impact, NTO sets sysctls via TuneD, not via MC's.

**Pull request merged**
